### PR TITLE
164 prepare platform layer and application for power supply firmware development

### PR DESCRIPTION
**Pull request by mlecriva**
- Adding base application for power supply control
- Adding platform layer for it with basic COGIP fw components:
  - CAN + protobuf messages
  - Heartbeat thread
  - Shell + Basic commands

cf. #164

### CODE REVIEW

File: platforms/pf-power-supply/include/platform.hpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+/**
+ * @brief Print current robot state in JSON format
+ *
+ * param[in]    out             tracefd descriptor used to print state
+ *
+ * @return
+ */
+void pf_print_state(void);
+
```
Comment (gdoffe) [inline platforms/pf-power-supply/include/platform.hpp:64]: you should remove this, not used here

File: platforms/pf-power-supply/include/platform.hpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+constexpr cogip::canpb::uuid_t copilot_connected_uuid = 0x3002;
+constexpr cogip::canpb::uuid_t copilot_disconnected_uuid = 0x3003;
+/** @} */
+
+/**
+ * @brief Get trace mode status
+ *
+ * @return                      true if trace mode is activited, false otherwise
+ */
+bool pf_trace_on(void);
```
Comment (gdoffe) [inline platforms/pf-power-supply/include/platform.hpp:47]: you should remove this, not used here

File: platforms/pf-power-supply/Makefile.dep [hunk 3]
```diff
@@ -1,10 +1,6 @@
 # RIOT modules
 USEMODULE += event
-USEMODULE += periph_gpio_irq
-USEMODULE += periph_qdec
-USEMODULE += motor_driver
 USEMODULE += printf_float
-USEMODULE += ps
 USEMODULE += scanf_float
```
Comment (gdoffe) [inline platforms/pf-power-supply/Makefile.dep:4]: I think you failed your commit squash :stuck_out_tongue:

File: applications/power-supply-control/include/app.hpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+/*
+ * Machine parameters
+ */
+
+#define USART_CONSOLE   USARTC0
```
Comment (gdoffe) [inline applications/power-supply-control/include/app.hpp:7]: not relevant here, should be in board

**Review by gdoffe — CHANGES_REQUESTED** [review]

File: platforms/pf-power-supply/Makefile.dep [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# RIOT modules
+USEMODULE += event
+USEMODULE += periph_gpio_irq
+USEMODULE += periph_qdec
+USEMODULE += motor_driver
```
[the remark this answers is not in the document]
Comment (mlecriva) [reply platforms/pf-power-supply/Makefile.dep]: fixed by resolving commit squash mistake.

File: platforms/pf-power-supply/Makefile.dep [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+USEMODULE += printf_float
+USEMODULE += ps
+USEMODULE += scanf_float
```
[the remark this answers is not in the document]
Comment (mlecriva) [reply platforms/pf-power-supply/Makefile.dep]: fixed !

File: platforms/pf-power-supply/Makefile.dep [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+USEMODULE += shell
+USEMODULE += shell_cmds_default
+
+# mcu-firmware modules
+USEMODULE += cogip_defs
+USEMODULE += canpb
+USEMODULE += utils
+
+# Board features required
+FEATURES_REQUIRED += periph_pwm
```
[the remark this answers is not in the document]
Comment (mlecriva) [reply platforms/pf-power-supply/Makefile.dep]: fixed by resolving commit squash mistake.

File: platforms/pf-power-supply/include/platform.hpp [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    platforms_pf-power-supply power supply control platform
+ * @ingroup     platforms
+ * @brief       COGIP power supply platform definition
+ *
+ * @author      Gilles DOFFE <g.doffe@gmail.com>
```
[the remark this answers is not in the document]
Comment (mlecriva) [reply platforms/pf-power-supply/include/platform.hpp]: fixed.

File: platforms/pf-power-supply/include/platform.hpp (hunk 1 above)
Comment (mlecriva) [reply platforms/pf-power-supply/include/platform.hpp]: fixed.

File: platforms/pf-power-supply/include/platform.hpp (hunk 2 above)
Comment (mlecriva) [reply platforms/pf-power-supply/include/platform.hpp]: fixed.

File: platforms/pf-power-supply/Makefile.dep (hunk 3 above)
Comment (mlecriva) [reply platforms/pf-power-supply/Makefile.dep]: yes ^^ Fixed.

File: applications/power-supply-control/include/app.hpp (hunk 4 above)
Comment (mlecriva) [reply applications/power-supply-control/include/app.hpp]: fixed.

**Comment by gdoffe** [comment]
@mlecriva please sign off your commits as it is is open source here :wink:

**Review by gdoffe — APPROVED** [review]
LGTM @mlecriva 

but please sign off your commits before merging as it is is open source here 😉

`Signed-off-by: Mathis Lécrivain <lecrivain.mathis@gmail.com>`

**Pull request merged**